Updating XM Cloud FAQ development section around migration-related questions

jst-cyr opened the pull request:
## Description / Motivation

Update to the FAQ to remove 2022 SYM reference and direct users to more current migration options.

Fixes #681

## How Has This Been Tested?

Local and on Vercel preview

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation update (non-breaking change; modified files are limited to the `/data` directory or other markdown files)

## Checklist:
- [X] I have read the Contributing guide.
- [X] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change.
- [X] My change is a documentation change and there are NO other updates required.
- [ ] My change has new or updated images which are stored in the `/public/images` folder that need to be migrated to Sitecore DAM

thomas-desmond commented on apps/devportal/data/markdown/pages/learn/faq/xm-cloud/development.md:
```diff
@@ -67,11 +67,31 @@ Sitecore also recommends as best practices:
+
+   This page on the Developer Portal lists out content related to migrating from the platform to the Composable DXP, including XM Cloud.
+
+1. [XM to XM Cloud content migration tool](/downloads/xm-cloud#xm-to-xm-cloud-content-migration-tool)
```
All numbers in code are "1" but on page they are showing properly as 1, 2, 3. Can code be updated to 1, 2, 3.

thomas-desmond commented on apps/devportal/data/markdown/pages/learn/faq/xm-cloud/development.md:
```diff
@@ -67,11 +67,31 @@ Sitecore also recommends as best practices:
+
+1. [XM to XM Cloud content migration tool](/downloads/xm-cloud#xm-to-xm-cloud-content-migration-tool)
+
+1. [Razl](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1002803)
```
All numbers in code are "1" but on page they are showing properly as 1, 2, 3. Can code be updated to 1, 2, 3.

thomas-desmond reviewed (changes requested):
I don't see how this is related to your code but the Table Of Contents does not update properly when you navigate to a new section of the FAQ. It's currently working in production. Is your branch behind `main`? 

<img width="942" alt="image" src="https://github.com/Sitecore/developer-portal/assets/24610108/53c93265-6765-477b-a455-b7154bdf4c8b">

jst-cyr commented:
> I don't see how this is related to your code but the Table Of Contents does not update properly when you navigate to a new section of the FAQ. It's currently working in production. Is your branch behind `main`?

@thomas-desmond : That was really odd! I updated from Main yesterday, but it looks like the fix in Main happened after that. I just updated the branch with the latest from Main and sure enough the issue resolved itself locally. I've pushed that up to the branch and it should be available for review soon.

thomas-desmond reviewed (approved):
Looks good now!
